tools/choco/tinytex/tools/chocolateyinstall.ps1

```diff
@@ -1,73 +1,25 @@
-﻿$ErrorActionPreference = 'stop'; # stop on all errors
- 
-$toolsDirActual  = "$(Split-Path -parent $MyInvocation.MyCommand.Definition)"
-$toolsDir = "$($env:TMP)\Texlive"
-$texliveInstall = "http://mirrors.ctan.org/systems/texlive/tlnet/install-tl.zip"
-$profileUrl = "https://yihui.org/gh/tinytex/tools/tinytex.profile"
-$pkgcustom = "https://yihui.org/gh/tinytex/tools/pkgs-custom.txt"
-$luatexmirror = "http://mirror.ctan.org/tex-archive/systems/texlive/tlnet/archive/luatex.win32.tar.xz"
-
-Get-WebFile `
--Url $texliveInstall `
--FileName "$toolsDir\\install-tl.zip"
-
-Get-ChocolateyUnzip `
--FileFullPath "$toolsDir\\install-tl.zip" `
--Destination "$toolsDir\\tinytex"
-
-#Remove the zip file
-Remove-Item "$toolsDir\\install-tl.zip"
-
-#download tinytex.profile and modify it (set texdir to ./TinyTeX)
-Get-WebFile `
--FileName "$toolsDir\\tinytex\\tinytex.profile" `
--Url $profileUrl
-
-(gc "$toolsDir\\tinytex\\tinytex.profile") -replace '\./', './TinyTex/' | Out-File -encoding ASCII "$toolsDir\\tinytex\\tinytex.profile"
-
-Add-Content -Path "$toolsDir\\tinytex\\tinytex.profile" -Value 'TEXMFCONFIG $TEXMFSYSCONFIG'
-Add-Content -Path "$toolsDir\\tinytex\\tinytex.profile" -Value 'TEXMFHOME $TEXMFLOCAL'
-Add-Content -Path "$toolsDir\\tinytex\\tinytex.profile" -Value 'TEXMFVAR $TEXMFSYSVAR'
-
-#download the custom package list
-Get-WebFile `
--FileName "$toolsDir\\tinytex\\pkgs-custom.txt" `
--Url $pkgcustom
-foreach ($c in (gc "$toolsDir\\tinytex\\pkgs-custom.txt").split()){
-    $pkgs="$pkgs $c"
+﻿$ErrorActionPreference = 'Stop';
+$toolsDir=Get-ToolsLocation
+$url        = 'https://yihui.org/tinytex/TinyTeX.zip'
+$packageArgs = @{
+  packageName   = $env:ChocolateyPackageName
+  unzipLocation = $toolsDir
+  url           = $url
+  checksum      = '2c51fa0d402dac3c78bf45d1adebadb21758403eef927b61f0a72c9cb00296d0'
+  checksumType  = 'sha256'
```

Comment:
I don't know how to calculate sha256 on Windows. Would md5 be okay? If not, I'll try to figure it out.

And can I store the checksum elsewhere? e.g. `https://yihui.org/tinytex/TinyTeX.zip.sha256` Then read it in this script. I'm asking because this checksum is likely to change from day to day, which means I'll have to update it frequently, but I don't want to update it in this script, because it will introduce new git commits on a daily basis.

Reply:
`md5` is ok no problem on that. The main problem is that once published it should not change. One thing we can do is add it as a release asset and let this script pull from that instead of the other one. 

That was the same problem faced with `install-tl.zip`, there also it was updated daily and checksum change. What the reviewer told was to use a constant file from `texlive.info` rather than getting checksums in runtime as you told. Once a month maybe we can update it. If you need I will write a GitHub Action for that. I don't think we need to update it daily.

Reply:
I meant I could provide a constant file like `https://yihui.org/tinytex/TinyTeX.zip.md5` which always stores the md5 checksum of the current TinyTeX.zip. Would that work? All I was asking was that whether it would be okay to pull the checksum from a URL instead of writing the string inside this `tinytex.nuspec`.

Reply:
> All I was asking was that whether it would be okay to pull the checksum from a URL instead of writing the string

I don't think they will allow. I previously did so for another chocolatey package and they told me not to do so. So what is actually required is a static file for a release that don't change.( You can add to GitHub Release assets). (We can update it once a month with a new release).

Reply:
Is the chocolatey package reviewed by a reviewer every time we make a new release? Or only the initial release?

Reply:
> Is the chocolatey package reviewed by a reviewer every time we make a new release? 

It's done everytime until we do some amount of releases till it become a trusted package.

Reply:
I see. In that case, I tend to create a new Github repo for this choco package, and update it monthly (could also be less or more frequently). We can write the checksum to that repo. Do you think this would be a better way to go?

Reply:
Yeah, looks good to me having a new repo. But anyhow they will be looking where we get the executables from(Package can be anywhere). So you create a redirect from your domain I think and have a mention about that in https://yihui.org/tinytex

Reply:
Also, can I push this to chocolatey? Looks like checksum changed today to `2c51fa0d402dac3c78bf45d1adebadb21758403eef927b61f0a72c9cb00296d0` . How can I make it constant?

Reply:
I'll create a new repo and stabilize the checksum. Basically I'll automate a monthly Github release. I think the choco reviewer should be happy with it.

Reply:
I may help you there if needed and I don't like delaying this up. Also, have a mention of this in https://yihui.org/tinytex as I have mentioned that as project page. So he don't ask again "what is relationship between the things you are getting from and the project and how to confirm?". But because of this I like chocolatey very much. ;-).

Reply:
I'll mention the download location on yihui.org/tinytex in a minute to address the reviewer's question.

Reply:
You can tell the reviewers that https://github.com/yihui/tinytex-releases contains official releases of TinyTeX now. This is reflected at: https://yihui.org/tinytex/#installation